### PR TITLE
Add `near generate-key` command

### PR DESCRIPTION
**Pull request by vgrichina**
Fixes https://github.com/nearprotocol/near-shell/issues/177

### CODE REVIEW

File: commands/generate-key.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    handler: exitOnError(async (argv) => {
+        let near = await require('../utils/connect')(argv);
+        if (argv.accountId) {
+            const { deps: { keyStore }} = near.config;
+            const keyPair = KeyPair.fromRandom('ed25519');
+            await keyStore.setKey(argv.networkId, argv.accountId, keyPair);
+            console.log(`Generated key pair with ${keyPair.publicKey} public key`);
+        }
+    })
+};
```
Comment (icerove) [inline commands/generate-key.js]: why do we have this function individually not with other functions? and this makes command and function together? why not follow what we already have?

Comment (vgrichina) [reply commands/generate-key.js]: @icerove didn't want to do too many chnages at once to pull every function, so for now only pulled new functions like this

The reason to do this is that it's more convenient to work with such structure if there are concurrent modifications that needs to be merged (e.g. we both add some commands).
Also current structure avoids extra step (you just define a command at once, without having to define a function first and then wrap it into command).

**Comment by vgrichina** [comment]
> Looks like this will override the old key. Is that the best option?

@bowenwang1996 do you think we should ask the user whether to overwrite old key? Or just no-op with warning if key does exists?

**Comment by bowenwang1996** [comment]
I think we should ask user before overriding the key file because once done, the private key would be gone. Normally that is fine because it's the same account, but if they actually want to use that key for a different network, they will lose their old key and that actually matters.

**Comment by vgrichina** [comment]
@bowenwang1996 the problem you described is solved by doing no-op with warning too though. Benefit over asking is that it's easier to put in a script (does not require user input).

BTW we also can add separate network ID for stake wars.

**Comment by bowenwang1996** [comment]
Sorry I don't understand. If you override your old key and that is your only key to your account, it seems that you will lose your account. Did I miss something?

**Comment by icerove** [comment]
How about only generate new public keys to that account and do not change account private key? not generate key pair but only public one. Since multi public keys are accepted.

**Comment by vgrichina** [comment]
> How about only generate new public keys to that account and do not change account private key? not generate key pair but only public one. Since multi public keys are accepted.

@icerove hmm, do you mean deriving multiple different key pairs from one private key? Sounds like a bit out of scope.

@bowenwang1996 
> Sorry I don't understand. If you override your old key and that is your only key to your account, it seems that you will lose your account. Did I miss something?

Yes, that I am suggesting to solve this problem in a different way: make near-shell not do anything when there is existing key for account, but just show a warning message. This also requires changes to PR, just a different change.

**Comment by bowenwang1996** [comment]
Oh I see. Sure that works.

**Pull request merged**